docstore/driver/util.go

```diff
@@ -61,8 +61,12 @@ func GroupActions(actions []*Action) (beforeGets, getList, writeList, afterGets
 	agets := map[interface{}]*Action{}
 	cgets := map[interface{}]*Action{}
 	writes := map[interface{}]*Action{}
+	var nilkeys []*Action
 	for _, a := range actions {
-		if a.Kind == Get {
+		if a.Key == nil {
```

Comment:
I believe Key is not necessarily nil, it could be empty string. Probably better check a.Kind == Create